### PR TITLE
DEPR: Remove Persistent Course Grades Flags

### PR DESCRIPTION
**Pull request by UvgenGen**
[OeX_Depr-288](https://youtrack.raccoongang.com/issue/OeX_Depr-288) [edx-platform] Remove Persistent Course Grades Flags

### CODE REVIEW

File: lms/djangoapps/courseware/tests/test_views.py [hunk 1]
```diff
@@ -1533,28 +1532,20 @@ def test_progress_queries_paced_courses(self, self_paced, query_count):
...
-        (True, 52, 36)
-    )
-    @ddt.unpack
-    def test_progress_queries(self, enable_waffle, initial, subsequent):
+    def test_progress_queries(self):
         ContentTypeGatingConfig.objects.create(enabled=True, enabled_as_of=datetime(2018, 1, 1))
         self.setup_course()
-        with override_waffle_switch(ASSUME_ZERO_GRADE_IF_ABSENT, active=enable_waffle):
+        with self.assertNumQueries(
+            52, table_ignorelist=QUERY_COUNT_TABLE_IGNORELIST
```
Comment (ruzniaievdm) [inline lms/djangoapps/courseware/tests/test_views.py:1539]: Why don't you use the second data pack? `(False, 60, 42)` instead used only one

Comment (ruzniaievdm) [inline lms/djangoapps/courseware/tests/test_views.py:1539]: Why don't you use the second data pack? `(False, 60, 42)` instead used only one

**Review by dyudyunov — APPROVED** [review]
Comments I left inline are more like places I would like to accent in the upstream PR. According to conversations with the edx team all looks fine. Approved.

File: lms/djangoapps/grades/config/__init__.py [hunk 2]
```diff
@@ -1,28 +1,3 @@
 """
```
[the remark this answers is not in the document]
Comment (UvgenGen) [reply lms/djangoapps/grades/config/__init__.py]: updated

File: lms/djangoapps/grades/subsection_grade_factory.py [hunk 3]
```diff
@@ -41,26 +40,25 @@ def create(self, subsection, read_only=False, force_calculate=False):
...
-        grade currently exists, even if the assume_zero_if_absent flag is enabled for the course.
+        grade currently exists.
         """
         self._log_event(
             log.debug, f"create, read_only: {read_only}, subsection: {subsection.location}", subsection,
         )
 
         subsection_grade = self._get_bulk_cached_grade(subsection)
         if not subsection_grade:
-            if assume_zero_if_absent(self.course_data.course_key) and not force_calculate:
```
Comment (idegtiarov) [inline lms/djangoapps/grades/subsection_grade_factory.py:52]: now `assume_zero_if_absent` is False by default could you confirm that the current condition works as aspected considering that first condition is False?

**Review by idegtiarov — CHANGES_REQUESTED** [review]
One question from me, waiting for the answers to Eugen's questions I'm queries in them as well.

**Review by idegtiarov — APPROVED** [review]
Missed your previous answer, it seems everything is fine and ready to upstream

**Pull request closed without merging**